blockchain/block/utils.go

```diff
@@ -19,20 +19,20 @@ import (
 	"github.com/iotexproject/iotex-core/crypto"
 )
 
-func calculateTxRoot(acts []action.SealedEnvelope) hash.Hash256 {
+func calculateTxRoot(acts []action.SealedEnvelope) (hash.Hash256, error) {
 	h := make([]hash.Hash256, 0, len(acts))
 	for _, act := range acts {
 		actHash, err := act.Hash()
 		if err != nil {
 			log.L().Debug("Error in getting hash", zap.Error(err))
-			return hash.ZeroHash256
+			return hash.ZeroHash256, err
 		}
 		h = append(h, actHash)
 	}
 	if len(h) == 0 {
-		return hash.ZeroHash256
+		return hash.ZeroHash256, errors.Errorf("hash length is zero")
```

Comment:
in the current code, if `CalculateTxRoot()` returns here, it is not considered an error

so i think to keep the same, should do `return hash.ZeroHash256, nil`, then don't need to add dummy action in `SignAndBuild()`

Reply:
Done. Removed dummy action from previous commit and changed logic in CalculateTxRoot. @dustinxie

Reply:
couple of more comments for formatting

Reply:
done. code pushed. @dustinxie